File: elixir/sublist/sublist.exs
defmodule Sublist do
  @doc """
  Returns whether the first list is a sublist or a superlist of the second list
  and if not whether it is equal or unequal to the second list.
  """
  def compare(a, b) do
    cond do
      a == b -> :equal
      is_sublist?(a, b) -> :sublist
      is_sublist?(b, a) -> :superlist
      true -> :unequal
    end
  end

  def is_sublist?(a, []), do: false
  def is_sublist?(a, [_|t] = b) do
    case List.starts_with?(b, a) do
      true -> true
      false -> is_sublist?(a, t)
    end
  end
end

File: python/sublist/sublist.py
SUBLIST = 0
SUPERLIST = 1
EQUAL = 2
UNEQUAL = 3


def check_lists(l1, l2):
    l1 = str(l1).replace('[', '').replace(']', '')
    l2 = str(l2).replace('[', '').replace(']', '')

    if l1 == l2:
        return EQUAL
    elif l1 in l2:
        return SUBLIST
    elif l2 in l1:
        return SUPERLIST
    return UNEQUAL
